pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/BrokersBase.java

```diff
@@ -614,5 +629,34 @@ private CompletableFuture<Void> doShutDownBrokerGracefullyAsync(int maxConcurren
         pulsar().getBrokerService().unloadNamespaceBundlesGracefully(maxConcurrentUnloadPerSec, forcedTerminateTopic);
         return pulsar().closeAsync();
     }
+
+
+    private CompletableFuture<Void> validateBrokerOperationAsync(String cluster, String brokerId,
+                                                                 BrokerOperation operation) {
+        final var pulsar = pulsar();
+        if (pulsar.getBrokerService().isAuthenticationEnabled()
+            && pulsar.getBrokerService().isAuthorizationEnabled()) {
+            return pulsar.getBrokerService().getAuthorizationService()
+                    .allowBrokerOperationAsync(cluster, brokerId, operation, originalPrincipal(),
+                            clientAppId(), clientAuthData())
+                    .thenAccept(isAuthorized -> {
+                        if (!isAuthorized) {
+                            throw new RestException(Status.UNAUTHORIZED,
+                                    String.format("Unauthorized to validateBrokerOperation for"
+                                                  + " originalPrincipal [%s] and clientAppId [%s] "
+                                                  + "about operation [%s] on broker [%s]",
+                                            originalPrincipal(), clientAppId(), operation.toString(), brokerId));
+                        }
+                    }).exceptionallyCompose(ex -> {
+                        // Fallback to superuser validation for compatibility purposes.
+                        final var rc = FutureUtil.unwrapCompletionException(ex);
+                        if (rc instanceof UnsupportedOperationException) {
```

Comment:
Here, we need to discuss if we can check `validateSuperUserAccessAsync` first.
@codelipenghui WDYT?

Reply:
fixed.